rdt/transformers/categorical.py

```diff
@@ -460,8 +462,16 @@ def _transform(self, data):
                 Data to transform.
 
         Returns:
-            numpy.ndarray:
+            pd.Series.
         """
+        unseen_categories = set(pd.unique(data)) - set(self.categories_to_values.keys())
+        if unseen_categories:
+            warn(
+                f'Warning: The data contains new categories {unseen_categories} that were not '
+                'seen in the original data. Assigning them NaN values. If you want to model '
+                'new categories, please fit the transformer again with the new data.'
+            )
+
         return pd.Series(data).map(self.categories_to_values)
 
     def _reverse_transform(self, data):
```

Comment:
The reverse transform already clips the data silently:

https://github.com/sdv-dev/RDT/blob/7848667a4768a43deaafbdac3e24c3c46ec895a6/rdt/transformers/categorical.py#L505